IpmiFeaturePkg/IpmiPowerRestorePolicy/IpmiPowerRestorePolicy.c

```diff
@@ -0,0 +1,87 @@
+/** @file IpmiPowerRestorePolicy.c
+
+  This file implementing a driver entrypoint that configure Power Restore Policy via IPMI command
+
+  Copyright (c) Microsoft Corporation.
+  PDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <PiPei.h>
+#include <Library/DebugLib.h>
+#include <Library/IpmiCommandLib.h>
+#include <Library/PlatformPowerRestorePolicyConfigurationLib.h>
+
+/**
+  Configure Power Restore Policy via IPMI
+
+  @param  FileHandle      Handle of the file
+  @param  PeiServices     List of possible PEI Services.
+
+  @return EFI_SUCCESS     Power Restore Policy is configured to expected setting successfully.
+  @return Others          Failed to configure Power Restore Policy
+**/
+EFI_STATUS
+EFIAPI
+IpmiPowerRestorePolicyEntry (
+  IN       EFI_PEI_FILE_HANDLE  FileHandle,
+  IN CONST EFI_PEI_SERVICES     **PeiServices
+  )
+{
+  EFI_STATUS                              Status;
+  IPMI_GET_CHASSIS_STATUS_RESPONSE        ChassisStatusRespData;
+  UINT8                                   CurrentPowerRestorePolicy;
+  UINT8                                   PlatformPowerRestorePolicy;
+  IPMI_SET_POWER_RESTORE_POLICY_REQUEST   SetRestorePolicyRequest;
+  IPMI_SET_POWER_RESTORE_POLICY_RESPONSE  SetRestorePolicyResponse;
+
+  //
+  // Get current power restore policy setting
+  //
+  Status = IpmiGetChassisStatus (&ChassisStatusRespData);
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "[%a] - IpmiGetChassisStatus: %r\n", __FUNCTION__, Status));
+    CurrentPowerRestorePolicy = POWER_RESTORE_POLICY_UNKNOWN;
+  } else {
+    CurrentPowerRestorePolicy = (ChassisStatusRespData.CurrentPowerState >> 5) & 0x3; // [6:5] - power restore policy
+  }
+
+  DEBUG ((DEBUG_INFO, "[%a] - CurrentPowerRestorePolicy: 0x%x\n", __FUNCTION__, CurrentPowerRestorePolicy));
+
+  //
+  // Get platform power restore policy setting
+  //
+  Status = GetPowerRestorePolicy (&PlatformPowerRestorePolicy);
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "[%a] - GetPowerRestorePolicy: %r\n", __FUNCTION__, Status));
+    return Status;
+  }
+
+  DEBUG ((DEBUG_INFO, "[%a] - PlatformPowerRestorePolicy: 0x%x\n", __FUNCTION__, PlatformPowerRestorePolicy));
+
+  if ( PlatformPowerRestorePolicy == CurrentPowerRestorePolicy) {
+    // Just return if CurrentPowerRestorePolicy the same with PlatformPowerRestorePolicy
+    DEBUG ((DEBUG_ERROR, "[%a] - Current Power Restore Policy is consistent with platform setting\n", __FUNCTION__));

```

Comment:
```suggestion
    DEBUG ((DEBUG_VERBOSE, "[%a] - Current Power Restore Policy is consistent with platform setting\n", __FUNCTION__));

```
Its not really an error, its just information that the system is configured as expected.